cli/tutors-json-manager/src/controller.ts

```diff
@@ -0,0 +1,77 @@
+import { encryptData, decryptData } from './encryption';
+import { compressData, decompressData } from './compression';
+import logger from './logger';
+
+interface EncryptedCompressedData {
+  isCompressed: boolean;
+  isEncrypted: boolean;
+  data: string;
+}
+
+export function compressAndEncrypt(data: object, compress: boolean = false, encrypt: boolean = false, key: string = ''): string {
+  try {
+    let jsonData = JSON.stringify(data);
+    let isCompressed = false;
+    let isEncrypted = false;
+
+    if (encrypt) {
+      if (!key) throw new Error('Encryption key must be provided');
+      jsonData = encryptData(jsonData, key);
+      isEncrypted = true;
+      logger.info('Encryption successful.');
+    }
+
+    if (compress) {
+      const originalSize = Buffer.byteLength(jsonData, 'utf8');
+      jsonData = compressData(jsonData);
+      isCompressed = true;
+      const compressedSize = Buffer.byteLength(jsonData, 'utf8');
+      const savedPercentage = ((originalSize - compressedSize) / originalSize) * 100;
+      logger.info(`Compression successful. Space saved: ${savedPercentage.toFixed(2)}%`);
+    }
+
+    return JSON.stringify({
+      isCompressed,
+      isEncrypted,
+      data: jsonData
```

Comment:
This is the structure we are left with meaning we can quickly get a true/false read on the JSON that is created and the "payload" of jsonData is then free to interact with and parse later.